src/experimental/information_estimates.cpp

```diff
@@ -0,0 +1,568 @@
+    m_obs_nu = nu;
+}
+
+InformationEstimates::InformationEstimates()
+{
+    m_max_sensor_range = 25.0;
+    m_cell_resol = 0.1;
+    m_obs_lambda = 0.35;
+    m_obs_nu = 0.28;
+}
+
+
+kt_double InformationEstimates::findMutualInfo(std::vector<karto::LocalizedRangeScan*> const& range_scans)
+{
+    /*
+        1. Parate en una celda de la grid inicial.
+        2. Identifica cuales laser scans del grupo actual pueden ver esta celda. (Maximum range radius)
+        3. Cuales haz de luz del scan seleccionado puede ver esa celda.
+        4. Resto de operaciones.
+    */
+
+    /*
+    Para el punto 0:
+        - Debe existir un loop exterior para ir removiendo un laser scan a la vez, con esto identifico cual es el scan
+        que aporta menor informacion del grupo dado.
+    */
+
+    /*
+    Para el punto 1 y 2:
+        - Toma el pose del scan actual.
+        - Encuentra la distancia del centro del scan con respecto al centro de la celda que nos interesa.
+        - Si esa diferencia es menor que el rango del scan, entonces consideramos este scan. (Agregamos ese scan a un vector)
+
+    Esto puedo afrontarlo de forma local, lo que significa que por cada set de scans que reciba,
+    puedo encontrar los extremos para poder reescalar el grid, esto potencialmente reduce el tiempo de busqueda.
+    */
+
+    /*
+    Para el punto 3:
+        - Con las lecturas de ese scan, encuentro el haz de luz que este dentro de la celda.
+        - Potencialmente puedo encontrar dos lectura, pero por ahora elegire el primero que encuentre.
+    */
+
+    // ------------ Start of Grid resizing ------------ //
+    // I took the first scan for reference. From this one I extract the min an max points (Poses)
+    m_low_x = range_scans[0]->GetCorrectedPose().GetX();
+    m_low_y = range_scans[0]->GetCorrectedPose().GetY();
+
+    m_high_x = range_scans[0]->GetCorrectedPose().GetX();
+    m_high_y = range_scans[0]->GetCorrectedPose().GetY();
+
+    //  Iterating through the group of scans to rescale the grid
+    for (const auto & scan : range_scans)
+    {
+        if (scan == nullptr)
+        {
+            continue;
+        }
+
+        karto::Pose2 pose = scan->GetCorrectedPose();
+        // std::cout << "Processing pose: " << pose.GetX() << ", " << pose.GetY() << std::endl;
+
+        // Finding the closest pose
+        m_low_x = pose.GetX() < m_low_x ? pose.GetX() : m_low_x;
+        m_low_y = pose.GetY() < m_low_y ? pose.GetY() : m_low_y;
+
+        // Finding the farthest pose
+        m_high_x = pose.GetX() > m_high_x ? pose.GetX() : m_high_x;
+        m_high_y = pose.GetY() > m_high_y ? pose.GetY() : m_high_y;
+    }
+
+    // Margins for the closest points
+    m_low_x -= m_max_sensor_range;
+    m_low_y -= m_max_sensor_range;
+
+    // Margins for the farthest points
+    m_high_x += m_max_sensor_range;
+    m_high_y += m_max_sensor_range;
+
+    // Map dimensions
+    kt_double dist_x = std::fabs(m_high_x) + std::fabs(m_low_x);
```

Comment:
@kmilo7204 hmm, shouldn't it be minus?
```suggestion
    kt_double dist_x = std::fabs(m_high_x) - std::fabs(m_low_x);
```
Same below.